docs/en_us/platform_api/source/courses/blocks.rst

```diff
@@ -0,0 +1,323 @@
+        "graded": false,
+        "student_view_url": "https://courses.edx.org/xblock/i4x://edX/DemoX/
+         course/Demo_Course",
+        "student_view_multi_device": false,
+        "lms_web_url": "https://courses.edx.org/courses/edX/DemoX/Demo_Course/
+          jump_to/i4x://edX/DemoX/course/Demo_Course",
+        "type": "course",
+        "id": "i4x://edX/DemoX/course/Demo_Course"
+        }
+    }
+ }
+
+
+.. _Get a list of the course blocks in a block tree:
+
+*********************************************
+Get a List of Course Blocks in a Block Tree
+*********************************************
+
+The endpoint to get a list of course blocks in a specified block tree is
+``/api/courses/v1/blocks/{usage_id}/``.
+
+=====================
+Use Case
+=====================
+
+Get a list of course blocks in a specified block tree. Response results depend
+on the requesting user's access level.
+
+=====================
+Example Requests
+=====================
+
+GET /api/courses/v1/blocks/{usage_id}/
+
+GET /api/courses/v1/blocks/i4x%3A%2F%2FedX%2FDemoX%2Fvertical
+%2F2152d4a4aadc4cb0af5256394a3d1fc7?all_blocks=true
+
+
+=====================
+Query Parameters
+=====================
+
+* all_blocks: (boolean) Provide a value of ``true`` to return all blocks.
+
+* block_counts: to return the aggregate number of blocks (including child
+  blocks) of each type specified in the "block_counts" response value.
+  Example: ``block_counts=video,problem``.
+
+* block_types_filter: Specify the types of blocks to be included in results.
+  Possible values are: ``sequential``, ``vertical``, ``html``, ``problem``,
+  ``video``   and ``discussion``. Example: ``block_types_filter=problem,html``.
+
+* depth: (integer or ``all``) Specify how far in the course blocks hierarchy
+  to traverse down. A value of ``all`` specifies the entire hierarchy. The
+  default value is ``0``. Example: ``depth=all``.
+
+* requested_fields: (list) Indicates fields to return for each block, in
+  addition to id, type, and display_name, which are always returned. For the
+  list of possible fields, see the fields listed under ``blocks`` in the
+  **Response Values** section below. Example:
+  ``requested_fields=graded,format,student_view_multi_device``.
+
+* return_type: (string) Indicates in what data type to return the block data.
+  Supported values are ``dict`` and ``list``. The default value is ``dict``.
+
+* student_view_data: (list) Indicates for which block types the JSON data is
+  returned in the "student_view_data" response value. Example:
+  ``student_view_data=video``.
+
+* usage_id: (string) The usage ID of the block. Example: ``i4x%3A%2F%2FedX
+  %2FDemoX%2Fvertical%2F2152d4a4aadc4cb0af5256394a3d1fc7``
+
+* username: (string) Required, unless ``all_blocks`` is specified. Example:
+  ``username=anjali``.
+
+
+=====================
+Response Values
+=====================
+
+The following fields are returned with a successful response.
```

Comment:
Can we ask the reader to refer to the previous endpoint's Response values, rather than repeating the text here?